internal/core/resolved.go

```diff
@@ -17,56 +17,66 @@ const (
 	ResolvedTCPRouteType  ResolvedRouteType = "TCPRoute"
 )
 
-// TODO Determine if Marshal needs to be here
 type ResolvedRoute interface {
 	GetType() ResolvedRouteType
 	GetMeta() map[string]string
 	GetName() string
 	GetNamespace() string
-	Marshal() ([]byte, error)
 }
 
-type marshaledRoute struct {
-	Type  ResolvedRouteType
-	Route json.RawMessage
+// resolvedRouteWrapper serves as a wrapper for a serialized ResolvedRoute.
+// We need this wrapper to indicate the type of route serialized in the Data
+// field so that it can be deserialized to the correct type.
+type resolvedRouteWrapper struct {
+	Type ResolvedRouteType
+	Data json.RawMessage
 }
 
-// TODO Is this the best location for this func?
-func UnmarshalRoute(b []byte) (ResolvedRoute, error) {
-	r := &marshaledRoute{}
-	if err := json.Unmarshal(b, r); err != nil {
+// UnmarshalResolvedRoute takes a serialized resolvedRouteWrapper and
+// deserializes it into the appropriate ResolvedRoute implementation.
+// It first partially deserializes to the wrapper in order to get the type
+// and then deserializes the Data field into the target HTTPRoute or TCPRoute.
```

Comment:
@sarahalsmiller @mikemorris we had discussed these functions a bit in #383 . This does seem like the best location IMO. I removed some redundant function wrappers and added godocs in this file to help explain what's going on and why it is this way.